### PR TITLE
Fixes to mia (OMA issue #482) 

### PR DESCRIPTION
**Pull request by ake123**
Fixes to mia (OMA issue #482) 

- diversity functions - warning that tells that some indices where not calculated 
- documentation  fix on runRDA()

### CODE REVIEW

File: R/estimateDiversity.R [hunk 1]
```diff
@@ -189,12 +189,17 @@
...
+#' valid_indices <- index[index %in% col_names]
+#' 
 #' # The colData contains the indices with their code names by default
-#' colData(tse)[, index]
+#' colData(tse)[, valid_indices]
 #'
 #' # Removing indices
-#' colData(tse)[, index] <- NULL
+#' colData(tse)[, valid_indices] <- NULL
+#' colData(tse)[, name] <- NULL
```
Comment (TuomasBorman) [inline R/estimateDiversity.R]: name was not yet used, so this line is extra (there cannot be index called name)

File: R/estimateDiversity.R [hunk 2]
```diff
@@ -240,229 +248,273 @@ NULL
...
-                index = c("coverage", "fisher", "gini_simpson", 
-                        "inverse_simpson", "log_modulo_skewness", "shannon"),
+           function(x, assay.type = "counts", assay_name = NULL,
+                    index = c("coverage", "fisher", "gini_simpson", 
+                              "inverse_simpson", "log_modulo_skewness", "shannon"),
                     name = index, ...)
-                    standardGeneric("estimateDiversity"))
+               standardGeneric("estimateDiversity"))
 
 #' @rdname estimateDiversity
```
Comment (TuomasBorman) [inline R/estimateDiversity.R]: Does this make the indentation wrong?

File: R/estimateDiversity.R [hunk 3]
```diff
@@ -240,229 +248,273 @@ NULL
...
+                       call. = FALSE)
+              }
+              .check_assay_present(assay.type, x)
+              .require_package("vegan")
+              
+              dvrsts <- BiocParallel::bplapply(index,
+                                               .get_diversity_values,
+                                               x = x,
+                                               mat = assay(x, assay.type),
+                                               BPPARAM = BPPARAM,
```
Comment (TuomasBorman) [inline R/estimateDiversity.R]: Only the indentation is changed. Also it seems that the indentation is not multipication of 4 spaces anymore

1. The indentation should be multiplication of 4
2. We do not want to change code like this. It looks like the whole code has changed. As people are already using the package, we must retain the code. It is extremely hard to see the history of code when there are this kind of changes that do not really affect the functionality. --> indentation is fixed only when the code/functionality is also changed

File: R/estimateDiversity.R (hunk 1 above)
Comment (ake123) [reply R/estimateDiversity.R]: removed ...was made during some testing

File: R/estimateDiversity.R (hunk 2 above)
Comment (ake123) [reply R/estimateDiversity.R]: i fixed it

File: R/estimateDiversity.R (hunk 3 above)
Comment (ake123) [reply R/estimateDiversity.R]: Okay fixed !

**Comment by ake123** [comment]
it only failed here https://github.com/microbiome/mia/actions/runs/9741095621/job/26879566844
It looks like some permission issue in the ubuntu

File: R/estimateDiversity.R [hunk 4]
```diff
@@ -286,183 +293,226 @@ setMethod("estimateDiversity", signature = c(x="SummarizedExperiment"),
...
+                  mat <- mat[ !is.na(node.label), ]
+                  node.label <- node.label[ !is.na(node.label) ]
+                  # Rename
+                  rownames(mat) <- node.label
+              }
+              # Calculates Faith index
+              faith <- list(.calc_faith(mat, tree, ...))
+              # Adds calculated Faith index to colData
+              .add_values_to_colData(x, faith, name)
+          }
```
Comment (TuomasBorman) [inline R/estimateDiversity.R]: It seems that the indentation is still off. Also, there are many lines that are not really changed. Only the indentation is changed. That makes review harder, but also makes the history of functions less clear.

Sorry about this, but can you revert lines that do not have any "real" changes that are not related to the issue?

Also, the permission issue in GHA should resolve when pushed to devel

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
